doc/configuration.rst

```diff
@@ -481,18 +481,21 @@ We recommend that you copy this set of exit codes to your ``luigi.cfg`` file:
   missing_data=20
   task_failed=30
   scheduling_error=35
+  unknown_reason=38
   unhandled_exception=40
 
 unhandled_exception
   For internal Luigi errors.  Defaults to 4, since this type of error
   probably will not recover over time.
+unknown_reason
+  For when a task fails or is left pending for unknown reasons.
```

Comment:
> Another reason for this could be workers forgotten by the scheduler? (from #1660)

Hmm, I'm not sure what @ulzha meant by "forgotten", I think the scheduler will not prune the task if it's something that was uploaded by the that's still alive worker.

> Perhaps a more correct (but generic) description could be: For when a task does not run successfully for unknown reasons

I like it! Let's not be specific or detailed, as that risks to quickly get out of date or be slightly incorrect to begin with.